### ISSUE TITLE
BERT binary head

### ISSUE THREAD

**Issue by qijiaxing** [qijiaxing/Megatron-LM#3]
In Megatron, there is an argument called `--bert-no-binary-head`. i.e. https://github.com/qijiaxing/Megatron-LM/blob/9d86ca677c48b4e006156b1edce631e6c4bd5c12/megatron/arguments.py#L343-L345

It can be used to disable the SOP/NSP loss in BERT pretrain.

Specifically, this argument is used in `pretrain_bert.py`, and it has two effects:
* It will set `num_tokentype = 0`, and further disable the `tokentype_embeddings` layer creation, and set the input of `tokentype_ids = None`
* It will disable the `Pooller` layer, i.e. `add_pooler = False`

https://github.com/qijiaxing/Megatron-LM/blob/9d86ca677c48b4e006156b1edce631e6c4bd5c12/pretrain_bert.py#L39-L44
https://github.com/qijiaxing/Megatron-LM/blob/9d86ca677c48b4e006156b1edce631e6c4bd5c12/megatron/model/language_model.py#L151-L159
https://github.com/qijiaxing/Megatron-LM/blob/9d86ca677c48b4e006156b1edce631e6c4bd5c12/pretrain_bert.py#L112-L113
https://github.com/qijiaxing/Megatron-LM/blob/9d86ca677c48b4e006156b1edce631e6c4bd5c12/megatron/model/language_model.py#L350-L351

To use such a model in finetune, we need to
* Disable the `Pooler` layer creation, see 6044dacb4bb696812d0eb18e50d50801fbaf79cd
* Disable the input of `tokentype_ids`, see 0190af76ae79748f9d4570c8b06c4e997f2da343

https://github.com/qijiaxing/Megatron-LM/blob/6044dacb4bb696812d0eb18e50d50801fbaf79cd/megatron/model/classification.py#L46-L54
https://github.com/qijiaxing/Megatron-LM/blob/6044dacb4bb696812d0eb18e50d50801fbaf79cd/megatron/model/classification.py#L88-L93
https://github.com/qijiaxing/Megatron-LM/blob/6044dacb4bb696812d0eb18e50d50801fbaf79cd/tasks/finetune_utils.py#L93-L94

**Comment by qijiaxing** [comment]
Did a pretrain on `new2016 small` dataset with SOP/NSP disabled, but found that it does NOT improve finetune tasks. 

Especially, `afqmc` become worse:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/jqi/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/jqi/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



Pretrain   Steps | tnews | afqmc | ocnli
-- | -- | -- | --
100 k | 55.77 | 69.0222 | 70.4407
150 k | 57.02 | 69.2539 | 70.339
200 k | 56.69 | 69.4393 | 71.661
250 k | 56.72 | 69.0454 | 71.7288
300 k | 56.31 | 68.9991 | 72.1017



</body>

</html>
